HW0/P5/P5.txt

```diff
@@ -1 +1,23 @@
-# Your answers here.
+# Qing Zhao (HUID: 80985053)’s answer for P5
+
+1. 255 seconds.
+
+2. The time for each of them to count 256 / 8 = 32 bags is 31 s; summing up 8 numbers takes at least 3s (2 ^ 3 = 8). Therefore, 8 cashiers can do this in 34 seconds.
+
+3. 8 seconds.
+
+4. See ‘P5.png’
+
+5. With infinite employees, it will take 8 + 7 = 15 seconds to count 256 bags;
```

Comment:
Actually you can do it in 8 seconds with 128 people I think. (256/128)-1+log_2(128) = 8 seconds